test/tool/net/uuidv4_test.lua

```diff
@@ -0,0 +1,23 @@
+-- Copyright 2022 Justine Alexandra Roberts Tunney
+--
+-- Permission to use, copy, modify, and/or distribute this software for
+-- any purpose with or without fee is hereby granted, provided that the
+-- above copyright notice and this permission notice appear in all copies.
+--
+-- THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+-- WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+-- WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE
+-- AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
+-- DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR
+-- PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER
+-- TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+-- PERFORMANCE OF THIS SOFTWARE.
+
+assert(#UuidV4() == 36)
+assert(string.sub(UuidV4(), 9, 9) == "-")
+assert(string.sub(UuidV4(), 14, 14) == "-")
+assert(string.sub(UuidV4(), 15, 15) == "4")
```

Comment:
I don't love having these tests run off of a single call to `UuidV4`. If there were an error, they would become flaky.

Maybe have the tests (at least the ones that check numeric values) run over a few iterations?

Reply:
I'm running one million tests now